### PR TITLE
Unified test documentation: nvdia support

### PR DESCRIPTION
**Pull request by sulewskiprzemyslaw**
Signed-off-by: Przemysław Sulewski <przemyslaw.sulewski@3mdeb.com>

### CODE REVIEW

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+```
+2d:00.0 3D controller: NVIDIA Corporation TU117M (rev a1)
+```
+
+### NVI001.002 NVIDIA Graphics detect (Windows 10)
+
+**Test description**
+
+This test aims to verify that the NVIDIA graphics card is initialized correctly
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:59]:
```suggestion
This test aims to verify that the NVIDIA graphics card is correctly initialized
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+1. Proceed with the
+    [Generic test setup: OS installation](../../generic-test-setup/#os-installation).
+1. Proceed with the
+    [Generic test setup: OS boot from disk](../../generic-test-setup/#os-boot-from-disk).
+
+### NVI001.001 NVIDIA Graphics detect (Ubuntu 20.04)
+
+**Test description**
+
+This test aims to verify that the NVIDIA graphics card is initialized correctly
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:23]:
```suggestion
This test aims to verify that the NVIDIA graphics card is correctly initialized
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+1. Proceed with the [Common](#common) section.
+1. Install the driver for the graphics card (GTX 1650) from [the official page](nvidia.com)
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
+1. Open the Start menu
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:77]:
```suggestion
1. Open the Start menu.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+1. Type in Device Manager
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:78]:
```suggestion
1. Type in `Device Manager`.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+1. Click on the Device Manager icon to open the Device Manager
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:79]:
```suggestion
1. Click on the Device Manager icon to open the Device Manager.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+1. Open the `Display adapters` section to wiew the GPUs present in the system
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:80]:
```suggestion
1. Open the `Display adapters` section to wiew the GPUs present in the system.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+**Test setup**
+
+1. Proceed with the [Common](#common) section.
+1. Install the driver for the graphics card (GTX 1650) from [the official page](nvidia.com)
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:76]:
```suggestion
1. Log into the system by using the proper login and password.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+
+**Test setup**
+
+1. Proceed with the [Common](#common) section.
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:39]:
```suggestion
1. Log into the system by using the proper login and password.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+
+```
+sudo apt install mesa-utils
+```
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:113]:
```suggestion
1. Log into the system by using the proper login and password.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+```
+
+1. Run the following command to see whether the card has turned on:
+
+```
+cat /sys/class/drm/card1/device/power/runtime_status
+```
+
+1. Close the test application and wait ~20 seconds to let the graphics card shut
+   itself down
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:135]:
```suggestion
   itself down.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+1. Run the following command to see whether the card has turned off again:
+
+```
+cat /sys/class/drm/card1/device/power/runtime_status
+```
+
+**Expected result**
+
+1. The output from the first command should be the word `suspended`
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:144]:
```suggestion
1. The output from the first command should be the word `suspended`.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+1. The output from the second command should be the word `active`
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:145]:
```suggestion
1. The output from the second command should be the word `active`.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 13]
```diff
@@ -0,0 +1,195 @@
...
+1. The output from the third command should be the word `suspended`
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:146]:
```suggestion
1. The output from the third command should be the word `suspended`.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+
+**Test configuration data**
+
+1. `FIRMWARE` = coreboot
+1. `OPERATING_SYSTEM` = Windows 10
+
+**Test setup**
+
+1. Proceed with the [Common](#common) section.
+1. Install the driver for the graphics card (GTX 1650) from [the official page](nvidia.com)
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:163]:
```suggestion
1. Install the driver for the graphics card (GTX 1650) from [the official page](nvidia.com).
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 15]
```diff
@@ -0,0 +1,195 @@
...
+1. Download and extract `gputest` from [Geeks3D](geeks3d.com/gputest)
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:164]:
```suggestion
1. Download and extract `gputest` from [Geeks3D](geeks3d.com/gputest).
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 16]
```diff
@@ -0,0 +1,195 @@
...
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:170]:
```suggestion
1. Log into the system by using the proper login and password.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 17]
```diff
@@ -0,0 +1,195 @@
...
+```
+2d:00.0 3D controller: NVIDIA Corporation TU117M (rev a1)
+```
+
+### NVI001.002 NVIDIA Graphics detect (Windows 10)
+
+**Test description**
+
+This test aims to verify that the NVIDIA graphics card is initialized correctly
+and can be detected by the operating system
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:60]:
```suggestion
and can be detected by the operating system.
```

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md [hunk 18]
```diff
@@ -0,0 +1,195 @@
...
+```
+sudo apt install mesa-utils
+```
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into system by using the proper login and password.
+1. Open a terminal window
```
Comment (damiankaras) [inline docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:114]:
```suggestion
1. Open a terminal window.
```

File: docs/unified-test-documentation/test-matrix.md [hunk 19]
```diff
@@ -3,7 +3,7 @@
 ## About
 
 <!--
-The test matrix is used to determine which of the test suites and test cases 
+The test matrix is used to determine which of the test suites and test cases
 described in this documentation are dedicated to the given platform
```
Comment (damiankaras) [inline docs/unified-test-documentation/test-matrix.md:7]:
```suggestion
described in this documentation are dedicated to the given platform.
```

**Review by damiankaras — CHANGES_REQUESTED** [review]
I left the review

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 2 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:23]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 3 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:77]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 5 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:79]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 6 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:80]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 7 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:76]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 10 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:135]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 11 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:144]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 12 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:145]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 14 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:163]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 15 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:164]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 16 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:170]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 17 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:60]: Improved

File: docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md (hunk 18 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/319-nvidia-graphics.md:114]: Improved

**Pull request merged**